protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepTools/PauseTools/index.tsx

```diff
@@ -1,5 +1,256 @@
+            gridGap={SPACING.spacing4}
+            width="100%"
+            padding={`${SPACING.spacing16} ${SPACING.spacing16} 0 ${SPACING.spacing16}`}
+          >
+            <RadioButton
+              onChange={(e: React.ChangeEvent<any>) => {
+                propsForFields.pauseAction.updateValue(e.currentTarget.value)
+              }}
+              buttonLabel={t(
+                'form:step_edit_form.field.pauseAction.options.untilResume'
+              )}
+              buttonValue={PAUSE_UNTIL_RESUME}
+              isSelected={
+                propsForFields.pauseAction.value === PAUSE_UNTIL_RESUME
+              }
+              largeDesktopBorderRadius
+            />
+            <RadioButton
+              onChange={(e: React.ChangeEvent<any>) => {
+                propsForFields.pauseAction.updateValue(e.currentTarget.value)
+              }}
+              buttonLabel={t(
+                'form:step_edit_form.field.pauseAction.options.untilTime'
+              )}
+              buttonValue={PAUSE_UNTIL_TIME}
+              isSelected={propsForFields.pauseAction.value === PAUSE_UNTIL_TIME}
+              largeDesktopBorderRadius
+            />
+            <RadioButton
+              onChange={(e: React.ChangeEvent<any>) => {
+                propsForFields.pauseAction.updateValue(e.currentTarget.value)
+              }}
+              buttonLabel={t(
+                'form:step_edit_form.field.pauseAction.options.untilTemperature'
+              )}
+              buttonValue={PAUSE_UNTIL_TEMP}
+              isSelected={propsForFields.pauseAction.value === PAUSE_UNTIL_TEMP}
+              largeDesktopBorderRadius
+              disabled={!pauseUntilModuleEnabled}
+            />
+          </Flex>
+          <Divider marginY="0" />
+          <Flex
+            flexDirection={DIRECTION_COLUMN}
+            gridGap={SPACING.spacing12}
+            paddingX={SPACING.spacing16}
+          >
+            {pauseAction === PAUSE_UNTIL_TIME ? (
+              <Flex
+                flexDirection={DIRECTION_COLUMN}
+                gridGap={SPACING.spacing12}
+              >
+                <Flex
+                  flexDirection={DIRECTION_COLUMN}
+                  gridGap={SPACING.spacing4}
+                >
+                  <StyledText desktopStyle="captionRegular">
+                    {t('application:time')}
+                  </StyledText>
+                  <StyledTextArea
+                    value={propsForFields.pauseTime.value as string}
+                    onChange={(e: React.ChangeEvent<any>) => {
+                      propsForFields.pauseTime.updateValue(
+                        e.currentTarget.value
+                      )
+                    }}
+                    error={
+                      propsForFields.pauseTime.errorToShow != null &&
+                      propsForFields.pauseTime.value != null &&
+                      propsForFields.pauseTime.value !== ''
+                    }
```

Comment:
i fixed a bug in my pr where the errors were getting populated before the field was dirty. I think that means for this, it can be simplified to just 
```suggestion
                    error={
                  errorToShow != null
                    }
```

Reply:
Do you mean changing the condition to only `propsForFields.pauseTime.errorToShow`? Leaving out `propsForFields.pauseTime.value` checks?